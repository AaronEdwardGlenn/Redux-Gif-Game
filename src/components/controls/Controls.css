.Controls {
  text-align: center;
  background-image: url(https://media2.giphy.com/media/hPlsN9Z6JNF4s/giphy.gif);
  background-size: cover;
  padding: 10px; 
  button {
    padding: 0.8rem;
    margin: 0 0.8rem;
    border: 2px solid yellow;
    background-color: green;
    color: #FFFFFF;
  }
  button:hover {
    background-color: #4CAF50; /* Green */
    color: white;
  }
}
