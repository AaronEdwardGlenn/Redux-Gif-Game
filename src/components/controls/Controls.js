import React from 'react';
import PropTypes from 'prop-types';
import styles from './Controls.css';

const Controls = ({ dispatch }) => (
  <section className={styles.Controls}>
    {dispatch.map(obj => (
      <button key={obj.text} onClick={obj.function}>
        {obj.text} {!!obj.count && `- ${obj.count}`}
      </button>
    ))}
  </section>
);

Controls.propTypes = {
  dispatch: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string.isRequired,
    count: PropTypes.number,
    function: PropTypes.func.isRequired,
  })).isRequired,

};

export default Controls;
