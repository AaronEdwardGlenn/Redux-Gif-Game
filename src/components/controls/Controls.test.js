import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls';

describe('Controls component', () => {
  it('can match the snap', () => {
    const wrapper = shallow(<Controls dispatch={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
