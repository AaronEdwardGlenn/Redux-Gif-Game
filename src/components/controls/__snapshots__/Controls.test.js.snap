// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component can match the snap 1`] = `
<section
  className="Controls"
/>
`;
